src/main.rs

```diff
@@ -107,16 +111,16 @@ pub fn start(mut os_input: Box<dyn OsApi>, opts: Opt) {
     let full_screen_ws = os_input.get_terminal_size_using_fd(0);
     os_input.into_raw_mode(0);
     let (send_screen_instructions, receive_screen_instructions): (
-        Sender<ScreenInstruction>,
-        Receiver<ScreenInstruction>,
+        Sender<(ErrorContext, ScreenInstruction)>,
```

Comment:
Could we have these be the other way around? (eg. `Sender<(ScreenInstruction, ErrorContext)>` I feel it will be clearer)